functions/consumer/sftp-consumer/src/main/java/org/springframework/cloud/fn/consumer/sftp/SftpConsumerSessionFactoryConfiguration.java

```diff
@@ -51,11 +49,8 @@ public SessionFactory<SftpClient.DirEntry> sftpSessionFactory(SftpConsumerProper
 		sftpSessionFactory.setPrivateKeyPassphrase(factory.getPassPhrase());
 		sftpSessionFactory.setAllowUnknownKeys(factory.isAllowUnknownKeys());
 		if (factory.getKnownHostsExpression() != null) {
-			// TODO probably a better way
-			String knownHosts = factory.getKnownHostsExpression()
-					.getValue(IntegrationContextUtils.getEvaluationContext(beanFactory), String.class);
-			ByteArrayResource knownHostsResource = new ByteArrayResource(knownHosts.getBytes(StandardCharsets.UTF_8));
-			sftpSessionFactory.setKnownHostsResource(knownHostsResource);
+			sftpSessionFactory.setKnownHostsResource(new FileSystemResource(factory.getKnownHostsExpression()
```

Comment:
This is doing exactly what the code was doing before in Spring Integration (the method that was removed) and I would prefer to not make any other changes in this effort. It does sound like it would be good to add though in a separate effort.

Reply:
Sorry, what was removed?
I just refer to this a a change which is not what was here before.
It was a `ByteArrayResource` and you changed it to `FileSystemResource`. Fully not the same.
So, where is Spring Integration problem here, please?

I'd like to help since I caused that big breaking in Spring Integration, but I need to understand the context of the change. 🤷

Reply:
Good point @artembilan - I did not add the context. 

So, originally to make things compile, [I switched to BAR](https://github.com/spring-cloud/stream-applications/commit/a8814f876f59616e12709eb65b271840c74d5209#diff-cee7856562f931c80686725cfb8a8eacd2851595c8942f28f3e9115bb95e65deR53) as the `setKnownHosts(String)` method was removed from SI. 

That temp fix did not work. The change in this PR makes it do what the corresponding removed method from SI used to do w/ the passed in string. 

In general we can expect main to be a bit in flux until we get to a bit more stability.

Reply:
Yeah.. OK. I see your point for the movement.
I guess if you'd ask me why did I do that change and what to do here, we wouldn't have this discussion 😄 

And now: do we have an issue to revise this code?
Because after this change we even don't have `TODO` any more, so the change you have just merged my stay there forever 😉

Reply:
> I guess if you'd ask me why did I do that change and what to do here, we wouldn't have this discussion 😄
Good point. You surely removed that code for a reason - it would be good to ask you why :) 

Is the main reason basically [this](https://github.com/spring-cloud/stream-applications/pull/347#discussion_r991446554)?

Reply:
If you ask about removal, then yes: `Resource` abstraction covers us almost for everything. If ask about the discussion then I got my answers with your explanation and commit history. I’ll probably come back tomorrow with some fix. Thank you for patience! 😅

Reply:
No, thank you for **your** patience my friend. I should have reached out and asked you about the removal to begin with. Thx for taking the time to come back through and review this.